### PR TITLE
Fix the import of external md files

### LINKED ISSUE #3458: Markdown shows up as HTML with addon-info
### Bug or support request summary

I'm using `@storybook/addon-info` to get some markdown into my stories, however, I'm loading it from an external file:

```js
// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import docs from './README.md';

const withDocs = withInfo(docs);

storiesOf('A story', module)
  .add('primary', withDocs(() => <p>Some JSX</p>))
```

What I'm noticing is that it works fine if the markdown just has a header:

```md
# Header
```

image

But as soon as I add another line, the info panel shows the first few lines as escaped HTML instead:

```md
# Header

some text
```

image

Any idea what I'm doing wrong?

### Steps to reproduce

1. make a vanilla react storybook: no custom webpack, etc.
2. install `@storybook/addon-info`
3. use this babelrc:

```
{
  "presets": [
    "babel-preset-env",
    "react",
    "flow"
  ],
  "plugins": [
    "transform-object-rest-spread",
    "transform-async-to-generator",
    "babel-plugin-styled-components"
  ]
}
```

4. Make a story:

```js
// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import docs from './README.md';

const withDocs = withInfo(docs);

storiesOf('A story', module)
  .add('primary', withDocs(() => <p>Some JSX</p>))
```

5. Add a header and a paragraph to the readme.md file.

### Please specify which version of Storybook and optionally any affected addons that you're running

- `@storybook/react` 3.4.2
- `@storybook/addon-info` 3.4.2

### Affected platforms 

Mac OS Sierra, Chrome 65

### PR DESCRIPTION
**Pull request by Keraito**
Issue: Fixes #3458. Right now, if markdown is imported from an external file, the content will be displayed as HTML. Different combinations of markdown elements would lead to different combinations of HTML and semi-proper markdown (styling was off). As far as I'm aware of from playing around with it, this is caused by auto conversion to HTML when importing the `.md` file.

## What I did
There were three options that I found that could work:
1. [Use webpack to change the import to load in the file raw.](https://github.com/rexxars/react-markdown/issues/76)
2. Automatically set the inner HTML. 
3. Turn the HTML back into proper markdown.

I didn't have experience with webpack and the second option ignored all the styling from existing components. So I implemented the last option with [turndown](https://github.com/domchristie/turndown). The condition for actually running the conversion is a regex checks on an HTML element at the start of the string. The option for the start was deliberate in order not to mess with markdown containing code examples.

## How to test
- Added 2 example stories, one with internal md and one with external.
- Added a snapshot test.

### CODE REVIEW

**Comment by Keraito** [comment]
Quick question: was I not supposed to commit the `.storyshot` and `.snap` files?

**Comment by Hypnosphi** [comment]
> was I not supposed to commit the .storyshot and .snap files?

You were

**Comment by Keraito** [comment]
@Hypnosphi, is there a reason the codecov report is not triggering?

**Comment by Hypnosphi** [comment]
IDK, but it's no big problem

**Comment by Hypnosphi** [comment]
BTW, I'm OK with replacing md-loader + html-loader with raw-loader, this looks like the easiest possible solution

**Comment by Keraito** [comment]
The tip of my local branch was behind on remote. Fixing it the way git tells me causes merge conflicts on code from one of the older commits (removed in later commit), so I decided to force push over it.

**Comment by Hypnosphi** [comment]
Looks like notes addon relies on old imports:

https://deploy-preview-3472--storybooks-official.netlify.com/?selectedKind=Addons%7CNotes&selectedStory=withNotes%20rendering%20imported%20markdown&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fnotes%2Fpanel

The easiest fix is to replace `{ notes: markdownNotes }` with `{ notes: { markdown: markdownNotes }}` in example, and adjust docs accordingly

**Comment by Keraito** [comment]
Will update the according documentation!

Regarding the notes addon, @Hypnosphi: While I see it in your link, I can't seem to reproduce it on my current `import-external-md` branch. 

screen shot 2018-04-28 at 16 09 25

**Comment by Hypnosphi** [comment]
What happens if you make a static build using `yarn build-storybook`?

**Comment by Keraito** [comment]
@Hypnosphi I have to run it from `examples/official-storybook` right? How do I check the static build after exactly?

**Comment by Hypnosphi** [comment]
Yes. You can open `storybook-static/index.html` in your browser manually or use some tool like [`html-server`](https://github.com/indexzero/http-server) for it

**Comment by Keraito** [comment]
So the markdown in the notes of https://github.com/storybooks/storybook/blob/851211019d2858a86c9bda2dde46919e36efa53d/examples/official-storybook/stories/addon-notes.stories.js#L35 actually works as intended (just like my previous schreenshot). But, [the story I added with an external md file](https://github.com/Keraito/storybook/blob/97402153494b45704b5f62b3d4bfd53eb5e7048e/examples/official-storybook/stories/addon-info.stories.js#L63-L66) does not. Did I mess up somewhere in the configs @Hypnosphi ?

**Comment by Hypnosphi** [comment]
That's weird, it works OK in [deployed version](https://deploy-preview-3472--storybooks-official.netlify.com/?selectedKind=Addons%7CInfo.Markdown&selectedStory=From%20external%20Markdown%20file&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fstories%2Fstories-panel)

Is there a chance you didn't run `yarn dev` or `yarn bootstrap --core`?

**Comment by Keraito** [comment]
Oh yes, now I reproduced it. I probably messed up something with the dependencies.. Will look into the notes addon and update the docs!

**Comment by Keraito** [comment]
Applied the requested changes @Hypnosphi

File: addons/notes/README.md [hunk 1]
```diff
@@ -62,11 +62,11 @@ import someMarkdownText from './someMarkdownText.md';
 storiesOf('Component', module).add(
   'With Markdown',
   () => <Component />
-  { notes: someMarkdownText }
+  { notes: { markdown: someMarkdownText } }
 );
 ```
 
-If you want to use Github flavored markdown inline, use `notes: { markdownText: 'your md' }`:
+Similarly, if you want to use Github flavored markdown inline, use `notes: { markdownText: 'your md' }`:
```
Comment (Hypnosphi) [inline addons/notes/README.md]: Looks like `markdownText` part is a typo in current docs, the example below uses `markdown`

Also, after your change there's no difference in handling inline vs imported markdown

**Review by Hypnosphi — APPROVED** [review]

**Comment by Hypnosphi** [comment]
Please update storyshot:

```
yarn test --core --update
```

**Comment by Keraito** [comment]
Done!

**Pull request merged**